File: app/src/main/java/es/etologic/mahjongscoring2/app/main/activity/MainActivityViewModelFactory.java
package es.etologic.mahjongscoring2.app.main.activity;

import android.arch.lifecycle.ViewModel;
import android.arch.lifecycle.ViewModelProvider;
import android.support.annotation.NonNull;

import es.etologic.mahjongscoring2.app.main.activity.MainActivityViewModel;

public class MainActivityViewModelFactory extends ViewModelProvider.NewInstanceFactory {

    @SuppressWarnings("unchecked")
    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) new MainActivityViewModel();
    }
}

File: app/src/main/java/es/etologic/mahjongscoring2/app/main/activity/MainActivityViewModelFactory.kt
package es.etologic.mahjongscoring2.app.main.activity

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import es.etologic.mahjongscoring2.domain.use_cases.*

class MainActivityViewModelFactory(
    private val createGameUseCase: CreateGameUseCase,
    private val updateCurrentGameUseCase: UpdateCurrentGameUseCase
) : ViewModelProvider.NewInstanceFactory() {
    
    @Suppress("UNCHECKED_CAST")
    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        return MainActivityViewModel(
            createGameUseCase,
            updateCurrentGameUseCase
        ) as T
    }
}
